rest_framework_simplejwt/authentication.py

```diff
@@ -16,25 +17,51 @@
 )
 
 
+def enforce_csrf(request):
+    """
+    Enforce CSRF validation.
+    """
+    check = CSRFCheck()
+    # populates request.META['CSRF_COOKIE'], which is used in process_view()
+    check.process_request(request)
+    reason = check.process_view(request, None, (), {})
+    if reason:
+        # CSRF failed, bail with explicit error message
+        raise exceptions.PermissionDenied('CSRF Failed: %s' % reason)
+
+
 class JWTAuthentication(authentication.BaseAuthentication):
     """
     An authentication plugin that authenticates requests through a JSON web
     token provided in a request header.
     """
     www_authenticate_realm = 'api'
 
+    def __init__(self):
+        self.user_model = get_user_model()
+
     def authenticate(self, request):
         header = self.get_header(request)
         if header is None:
-            return None
+            if not api_settings.AUTH_COOKIE:
+                return None
+            raw_token = request.COOKIES.get(api_settings.AUTH_COOKIE) or None
+        else:
+            raw_token = self.get_raw_token(header)
 
-        raw_token = self.get_raw_token(header)
         if raw_token is None:
             return None
 
         validated_token = self.get_validated_token(raw_token)
 
-        return self.get_user(validated_token), validated_token
+        user = self.get_user(validated_token)
+        if not user or not user.is_active:
```

Comment:
I think this is now handled by the new auth rules (we definitely need to update our changelog and documentation)...